pkgs/by-name/go/google-chrome/package.nix

```diff
@@ -159,116 +168,202 @@ let
+
+      substituteInPlace $out/share/google/$appname/google-$appname \
+        --replace-fail 'CHROME_WRAPPER' 'WRAPPER'
+      substituteInPlace $out/share/applications/google-$appname.desktop \
+        --replace-fail /usr/bin/google-chrome-$dist $exe
+      substituteInPlace $out/share/gnome-control-center/default-apps/google-$appname.xml \
+        --replace-fail /opt/google/$appname/google-$appname $exe
+      substituteInPlace $out/share/menu/google-$appname.menu \
+        --replace-fail /opt $out/share \
+        --replace-fail $out/share/google/$appname/google-$appname $exe
+
+      for icon_file in $out/share/google/chrome*/product_logo_[0-9]*.png; do
+        num_and_suffix="''${icon_file##*logo_}"
+        if [ $dist = "stable" ]; then
+          icon_size="''${num_and_suffix%.*}"
+        else
+          icon_size="''${num_and_suffix%_*}"
+        fi
+        logo_output_prefix="$out/share/icons/hicolor"
+        logo_output_path="$logo_output_prefix/''${icon_size}x''${icon_size}/apps"
+        mkdir -p "$logo_output_path"
+        mv "$icon_file" "$logo_output_path/google-$appname.png"
+      done
+
+      makeWrapper "$out/share/google/$appname/google-$appname" "$exe" \
+        --prefix LD_LIBRARY_PATH : "$rpath" \
+        --prefix PATH            : "$binpath" \
+        --suffix PATH            : "${lib.makeBinPath [ xdg-utils ]}" \
+        --prefix XDG_DATA_DIRS   : "$XDG_ICON_DIRS:$GSETTINGS_SCHEMAS_PATH:${addDriverRunpath.driverLink}/share" \
+        --set CHROME_WRAPPER  "google-chrome-$dist" \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+        --add-flags ${lib.escapeShellArg commandLineArgs}
+
+      for elf in $out/share/google/$appname/{chrome,chrome-sandbox,chrome_crashpad_handler}; do
+        patchelf --set-rpath $rpath $elf
+        patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $elf
+      done
+
+      runHook postInstall
+    '';
+
+    passthru.updateScript = lib.getExe (writeShellApplication {
+      name = "google-chrome-update-script";
+      runtimeInputs = [
+        curl
+        jq
+        common-updater-scripts
+      ];
+      text = ''
+        url="https://versionhistory.googleapis.com/v1/chrome/platforms/linux/channels/stable/versions/all/releases"
+        response="$(curl --silent --fail $url)"
+        version="$(jq ".releases[0].version" --raw-output <<< $response)"
+        if [[ "${finalAttrs.version}" = "$version" ]]; then
+            echo "The new version same as the old version."
+            exit 0
+        fi
+        update-source-version "google-chrome" "$version"
+      '';
+    });
+  });
+
+  darwin = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit pname meta;
+    version = "127.0.6533.89";
+
+    src = fetchurl {
+      url = "http://dl.google.com/release2/chrome/aclo5k72baz7erlbweyjruguwh2a_127.0.6533.89/GoogleChrome-127.0.6533.89.dmg";
```

Comment:
Uhhh

Reply:
I can't actually say for sure because querying old versions on the XML POST request always gives you the latest version, unless I'm doing it wrong... So I would imagine it changes each time

P.S: I notice something interesting tho both Windows x64 and macOS arm64 have almost same hash?

```console
# Windows x32
oct5rowwm5fvvxipazq2wlma2u

# Windows x64
acljsqmj242ipgr5z4ha6ymv5ita

# macOS
aclo5k72baz7erlbweyjruguwh2a
```

Reply:
Could you move the update script to a seperate update.sh (for both Linux and Darwin)? The derivation is way too large as it is.

Reply:
and would be nice if the darwin package could be updated from linux from using the same update script (obviously we can't run them though). This will make sure it stays updated by ryantm-r (which only runs on x64-linux AFAIK) if we run behind the updates.

And if you want to maintain this, do add yourself to maintainers list.

Reply:
> Could you move the update script to a separate update.sh (for both Linux and Darwin)? The derivation is way too large as it is.

Sure! But could you give me some "good" examples of ./update.sh scripts? I was looking at the manual, but it only mentions it was possible without any examples.

I was looking at the repo for update.sh files, but each one is pretty different

> And if you want to maintain this, do add yourself to meta.maintainers.

I don't use Chrome at all, so I won't really be a good maintainer for it :/

Reply:
> I don't use Chrome at all, so I won't really be a good maintainer for it :/

K, as long as running

```
nix-shell maintainers/scripts/update.nix --argstr package google-chrome
```
updates both linux and darwin packages I have no issues with it having no darwin maintainers.

I don't think this should be any much of an effort at all, given we are just unpacking and installing the dmg file. I don't have a darwin system so won't be able to test if it runs properly though.

Reply:
> Sure! But could you give me some "good" examples of ./update.sh scripts? I was looking at the manual, but it only mentions it was possible without any examples.

You could move the linux part to `update_linux ()` bash function. The same goes for `update_darwin ()`. The main part should call the both functions.

https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/editors/android-studio/update.sh

This could be helpful.